### PR TITLE
Add specialization constant composite test

### PR DESCRIPTION
**Pull request by eragoh**

### CODE REVIEW

File: conformance_tests/core/test_module/src/test_module.cpp [hunk 1]
```diff
@@ -343,6 +345,59 @@ TEST_F(
...
+  }
+
+  lzt::free_memory(buff_spec);
+  lzt::destroy_module(module_spec);
+}
+
+TEST_F(
+    zeModuleCreateTests,
+    GivenValidSpecConstantsThenUsingSpecConstantCompositeExpectVectorCreates) {
+  RunGivenSpecConstantsUsingSpecConstantCompositeWhenCreatingModuleTest(false);
```
Comment (Jemale) [inline conformance_tests/core/test_module/src/test_module.cpp:392]: could you add a 'When' clause in the test name to conform to the Given_When_Then format?

File: conformance_tests/core/test_module/src/test_module.cpp [hunk 2]
```diff
@@ -343,6 +345,59 @@ TEST_F(
...
+}
+
+TEST_F(
+    zeModuleCreateTests,
+    GivenValidSpecConstantsThenUsingSpecConstantCompositeExpectVectorCreatesOnImmediateCmdList) {
```
Comment (Jemale) [inline conformance_tests/core/test_module/src/test_module.cpp:397]: also edit the name here

Comment (eragoh) [reply conformance_tests/core/test_module/src/test_module.cpp]: Done

**Review by Jemale — APPROVED** [review]

**Comment by Jemale** [comment]
@eragoh I overlooked this, but approved just now, thanks!

**Pull request merged**